### ISSUE TITLE
Late joiners to multiparty networks may hang processing earlier definitions

### ISSUE THREAD

**Issue by awrichar** [hyperledger/firefly#1397]
When a node receives a definition broadcast (such as an FFI), it checks the local org definition to determine if the broadcast was sent by the local node or another:

https://github.com/hyperledger/firefly/blob/c454f8dca4e65b5891f6507da5dfa8aa3a3075d7/internal/definitions/handler_contracts.go#L164

However, if the local org does not exist in the database, this is interpreted as an error:

https://github.com/hyperledger/firefly/blob/c454f8dca4e65b5891f6507da5dfa8aa3a3075d7/internal/identity/identitymanager.go#L124

As a result, a late joiner to the network will fail to process definitions sent before it joined.

**Comment by peterbroadhurst** [comment]
Looks like the "so that" for why we need the check is here:
https://github.com/hyperledger/firefly/blob/c454f8dca4e65b5891f6507da5dfa8aa3a3075d7/internal/definitions/handler_contracts.go#L90-L97

**Comment by peterbroadhurst** [comment]
So a bit of analysis is needed into whether that check is critical.
If `existing.Message == nil` then clearly it does not matter if we don't have a local org registered.

If `existing.Message != nil` and we don't have a registered local org yet, then my assumption without doing the extra thinking cycles, is it's fine to process it as if we were not the author. The thinking being that we can't be the author or something published before we registered ourselves (understanding that's a bit of a simplification given topics mean ordering between those events is not necessarily guaranteed)

**Comment by awrichar** [comment]
I think actually we can just do a string comparison on the author DID. We should know our own DID, regardless of whether we've processed the registration or not. And by the time we get to the definition handler, we've already verified that the message came from a registered org, because the on-chain consistency check now happens quite early in aggregation:

https://github.com/hyperledger/firefly/blob/c454f8dca4e65b5891f6507da5dfa8aa3a3075d7/internal/events/aggregator.go#L447C36-L449

So knowing that the message came from a registered identity, and then checking if that identity matches the local org DID is enough. No need to _look up_ the local org again at this step either way.
